python/ql/lib/semmle/python/frameworks/Stdlib.qll

```diff
@@ -277,6 +281,13 @@ module Stdlib {
       ClassInstantiation() {
         this = subclassRef().getACall()
         or
+        this =
+          DD::selfTracker(subclassRef()
+                .getAValueReachableFromSource()
+                .asExpr()
+                .(ClassExpr)
+                .getInnerScope())
+        or
```

Comment:
could you add a test indicating we can now do this?

Reply:
Ah good point...this is one bit I am somewhat unsure of, do we want this? Also, it should trigger a performance check.

Reply:
I see you added this in https://github.com/github/codeql/pull/16840/commits/77a00873a974c4ecec229f19a59f2cef7ede0c02 -- I think a DCA performance check for this whole PR is in place yes :+1:

Reply:
I have set off a standard nightly/nightly DCA run. I did not include any extra options, so it should still extract the standard library; thus we should see a slight slowdown from the new modelling and we can gauge that it is not huge. (If we did stop extraction, that speedup would drown out what we are looking for.)